collated/main/A0139498J.md

```diff
@@ -0,0 +1,1044 @@
+     */
+    public static Priority convertStringToPriority(String priorityString) {
+        assert priorityString != null;
+        
+        switch (priorityString) {
+        case ("low"):
+        case ("l"):
+            return Priority.LOW;
+        case ("high"):
+        case ("h"):
+            return Priority.HIGH;
+        case ("medium"):
+        case ("med"):
+        case ("m"):
+        default:
+            return Priority.MEDIUM;
+        }
+    }
+}
+```
+###### \java\seedu\address\model\item\ReadOnlyTask.java
+``` java
+public interface ReadOnlyTask {
+
+    Name getName();
+    Priority getPriorityValue();
+    Optional<Date> getStartDate();
+    Optional<Date> getEndDate();
+    Optional<RecurrenceRate> getRecurrenceRate();
+    
+    /**
+     * Returns true if both have the same state. (interfaces cannot override .equals)
+     */
+    default boolean isSameStateAs(ReadOnlyTask other) {
+        return other == this // short circuit if same object
+                || (other != null // this is first to avoid NPE below
+                && other.getName().name.equals(this.getName().name) // state checks here onwards
+                && other.getPriorityValue() == this.getPriorityValue())
+                && other.getStartDate().equals(this.getStartDate())
+                && other.getEndDate().equals(this.getEndDate());
+    }
+    
+    /**
+     * Formats the floating task as text, showing all details.
+     */
+    default String getAsText() {
+        final StringBuilder builder = new StringBuilder();
+        builder.append(getName().name)
+                .append(", Priority: ")
+                .append(getPriorityValue());
+        if (getStartDate().isPresent()) {
+            builder.append(", StartDate: ").append(getStartDate().get());
+        }
+        if (getEndDate().isPresent()) {
+            builder.append(", EndDate: ").append(getEndDate().get());
+        }
+        return builder.toString();
+    }
+    
+}
+```
+###### \java\seedu\address\model\item\Task.java
+``` java
+    public Task(Name taskName) {
+        this.taskName = taskName;
+        this.priority = Priority.MEDIUM;
+    }
+
+```
+###### \java\seedu\address\model\item\Task.java
+``` java
+    public Task(Name taskName, Priority priorityValue) {
+        this.taskName = taskName;
+        this.priority = priorityValue;
+    }
+
+    /**
+     * Copy constructor.
+     * 
+     * @throws IllegalValueException
+     */
+    public Task(ReadOnlyTask source) {
+        Date tempStartDate = null, tempEndDate = null;
+        if (source.getStartDate().isPresent()) {
+            tempStartDate = source.getStartDate().get();
+        }
+        if (source.getEndDate().isPresent()) {
+            tempEndDate = source.getEndDate().get();
+        }
+
+        this.taskName = source.getName();
+        this.startDate = tempStartDate;
+        this.endDate = tempEndDate;
+        if (source.getRecurrenceRate().isPresent())
```

Comment:
coding standard violations; need braces for even the single statement conditional blocks